lib/iris/tests/__init__.py

```diff
@@ -695,18 +704,38 @@ def check_graphic(self, tol=_DEFAULT_IMAGE_TOLERANCE):
 
             figure.savefig(result_fname)
 
-            if not os.path.exists(expected_fname):
-                warnings.warn('Created image for test %s' % unique_id)
-                shutil.copy2(result_fname, expected_fname)
+            # XXX: Deal with a new test result i.e. it's not in the repo
 
-            err = mcompare.compare_images(expected_fname, result_fname, tol=tol)
+            # hash the created image using sha1
+            with open(result_fname, 'rb') as res_file:
+                sha1 = hashlib.sha1(res_file.read())
+            if unique_id not in repo:
+                msg = 'Image comparison failed: Created image {} for test {}.'
+                raise ValueError(msg.format(result_fname, unique_id))
```

Comment:
@marqh I completely understand your concerns, and I think that there is a suitable compromise to be had here to meet all our needs.

I don't think it would be wise of me to mix this into this PR - let's do that in a separate PR :+1: